### ISSUE TITLE
AbstractBundle.prepareContainer() cannot find "plugin.xml" if resolved path contain spaces

### ISSUE THREAD

**Issue by claesrosell** [eclipse-cdo/cdo#74]
This problem seems to have been introduced when the deprecated java.net.URL() construct was replaced with URI.create(string).toURL()

This is a slightly compressed stacktrace from a locally built version of CDO ( from the WSS PR )

```
java.lang.IllegalArgumentException: Illegal character in opaque part at index 17: jar:file:/C:/path to workspace/my_workspace/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/org.eclipse.emf.cdo.server.net4j_4.7.1.v20240801-1126.jar!/plugin.xml
	at java.base/java.net.URI.create(URI.java:932)
	at org.eclipse.net4j.util.io.IOUtil.newURL(IOUtil.java:224)
	at org.eclipse.net4j.internal.util.bundle.AbstractBundle.getInputStream(AbstractBundle.java:500)
	at org.eclipse.net4j.internal.util.bundle.AbstractBundle.prepareContainer(AbstractBundle.java:152)
	at org.eclipse.emf.cdo.server.net4j.CDONet4jServerUtil.prepareContainer(CDONet4jServerUtil.java:39)
....
Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 17: jar:file:/C:/path to workspace/my_workspace/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/org.eclipse.emf.cdo.server.net4j_4.7.1.v20240801-1126.jar!/plugin.xml
	at java.base/java.net.URI$Parser.fail(URI.java:2995)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3166)
	at java.base/java.net.URI$Parser.parse(URI.java:3202)
	at java.base/java.net.URI.<init>(URI.java:645)
	at java.base/java.net.URI.create(URI.java:930)
	... 23 common frames omitted
```
The exception happens in `org.eclipse.net4j.util.io.IOUtil.newURL(String)` but the URL-string is created earlier. Here comes my investigation:
While the now deprecated `new java.net.URL(String)` allowed white spaces in the input, `URI.create(String)` does not.

The URL string is created with data from `org.eclipse.net4j.internal.util.bundle.AbstractBundle.getBaseURL()`. That method can return unescaped URLs since it uses `FileLocator.resolve()` which gladly returns white space URLs (`org.eclipse.osgi.storage.url.BundleURLConverter` still uses the URL construct)

I think that if  `org.eclipse.net4j.util.io.IOUtil.newURL()` is changed to use `org.eclipse.core.runtime.URIUtil.fromString(url).toURL()` instead of `java.net.URI.create(url).toURL()` if think that it will solve this problem.
I will do some more tests and, if successfull, I will provide a PR

**Comment by claesrosell** [comment]
Fixed by 9f44bb4181ae7641faebd212ada08616b110d8ac

**Comment by merks** [comment]
@claesrosell 

Thanks for spotting these problems before they end up in the release tomorrow!  ❤️ 🏆
